zinc/src/main/scala/sbt/internal/inc/javac/AnalyzingJavaCompiler.scala

```diff
@@ -158,10 +183,15 @@ final class AnalyzingJavaCompiler private[sbt] (
       timed(javaAnalysisPhase, log) {
         for ((classesFinder, oldClasses, srcs) <- memo) {
           val newClasses = Set(classesFinder.get: _*) -- oldClasses
-          Analyze(newClasses.toSeq, srcs, log)(callback, loader, readAPI)
+          Analyze(newClasses.toSeq, srcs, log, output, finalJarOutput)(callback, loader, readAPI)
         }
       }
 
+      loader match {
+        case u: URLClassLoader => u.close()
```

Comment:
What's the purpose of this new logic?

Reply:
Well. We are constructing loader `val loader = ClasspathUtilities.toLoader(searchClasspath)` like that, but it is never closed, hodling the jars that it accessed causing problems when we write to them.

Reply:
Let's move this and the other change to a new commit and explain why they are used to (this is a general-purpose issue that affects also jars in the classpath that come from dependency resolution).